pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -555,6 +555,17 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		networks[network.Name] = network.DeepCopy()
 	}
 
+	if vmi.Spec.Domain.IOThreads != nil {
```

Comment:
does libvirt apply it automatically to relevant disk and controllers?
for `virtion-iscsi` iothread must be included under controller
```
 <controller type='scsi' index='0' model='virtio-scsi'>
    <driver iothread='0'/>
    <address type='pci' domain='0x0000' bus='0x00' slot='0x0b' function='0x0'/>
  </controller>
```
and for `virtio` under the disk
```
    <disk type='file' device='disk' snapshot='no'>
      <driver name='qemu' type='qcow2' cache='none' error_policy='stop' io='threads' iothread='1'/>
      ...
      <backingStore/>
      <target dev='vda' bus='virtio'/>
      <serial>f286c97f-720f-4eec-a511-b07791a6f065</serial>
      <boot order='1'/>
      <alias name='ua-f286c97f-720f-4eec-a511-b07791a6f065'/>
    </disk>
```

Reply:
This is `only` if we want to specify the exact IOThread we want to place this disk on.
I don't think that there is a need for this in our case.

Reply:
you sure? I thought it wont' get assigned/used unless you link it to the disk

Reply:
hmm, iirc it's not needed, but I'll double check.
If it is needed, that would complicate things. If there are more than 1 IOThreads we will need to account and decide which device is assigned to which thread. Disk hotplug will be a problem as well.

Reply:
Thanks for the insight @cynepco3hahue, I didn't realize I needed that.

@vladikr, @michalskrivanek there was a completely different approach I toyed with at first. After thinking on your feedback, I'm not sure it wasn't the better way:

Instead of listing an "iothreads" count as a toplevel stanza, we could instead add two booleans to each disk device: `useIOThreads` and `dedicatedIOThread`. I decided against it because I didn't think the libvirt XML was expressive enough to allow those directives to be translated in an explicit and correct way.

The above also gets a little bit cumbersome when we start talking about pinning iothreads. But food for thought. Is that approach viable or horrible? :)